src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Com.cs

```diff
@@ -68,7 +71,17 @@ public static int FinalReleaseComObject(object o)
         [SupportedOSPlatform("windows")]
         public static IntPtr GetComInterfaceForObject(object o, Type T)
         {
-            throw new PlatformNotSupportedException(SR.PlatformNotSupported_ComInterop);
+            if (o is null)
+            {
+                throw new ArgumentNullException(nameof(o));
+            }
+
+            if (T is null)
+            {
+                throw new ArgumentNullException(nameof(T));
+            }
+
+            return ComWrappers.ComInterfaceForObject(o, new Guid(T.GetCustomAttribute<GuidAttribute>().Value));
```

Comment:
Why not just call `Type.GUID` here? `GetCustomAttribute` is not going to be any faster than `Type.GUID`.